src/aiai_eval/evaluator.py

```diff
@@ -149,10 +153,16 @@ def evaluate(
         # Evaluate all the models in `model_ids` on all the datasets in `dataset_tasks`
         for task_config in task_configs:
             for m_id in model_ids:
-                self._evaluate_single(
-                    task_config=task_config,
-                    model_id=m_id,
-                )
+                try:
+                    self._evaluate_single(
+                        task_config=task_config,
+                        model_id=m_id,
+                    )
+                except InvalidArchitectureForTask:
+                    logger.warning(
+                        f"Skipping evaluation of {m_id} on {task_config.pretty_name} "
+                        "as the architecture is not supported by the task."
+                    )
```

Comment:
There's already a try-except block within `_evaluate_single`, which currently captures `InvalidEvaluation` - this should probably be moved in there.

We could _either_ catch separate exceptions within that method, _or_ make a base Exception class from which all of our custom Exceptions inherit from, and catch that exception in `_evaluate_single`.

Reply:
I did the former.